mfp_functions/fetchSingleDate.js

```diff
@@ -27,14 +27,45 @@ var fetchSingleDate = function(username, date, fields, callback){
     var cols = {};
 
     //find and set column numbers of nutrient fields
-    $('#food').find('thead').find('tr').find('td').each(function(index, element){
+    $('#food').find('thead').find('tr').find('td').each(function (index, element) {
       var $element = $(element);
       var fieldName = $element.text().toLowerCase();
-      if (fieldName === "sugars") { fieldName = "sugar"; } // fixes MFP nutrient field name inconsistency
-      if (fieldName === "cholest") { fieldName = "cholesterol"; } // fixes MFP nutrient field name inconsistency
-      if (index !== 0) { cols[fieldName] = index; } //ignore first field, which just says "Foods"
+      if (fieldName === "sugars") {
+        fieldName = "sugar";
+      } // fixes MFP nutrient field name inconsistency
+      if (fieldName === "cholest") {
+        fieldName = "cholesterol";
+      } // fixes MFP nutrient field name inconsistency
+      if (index !== 0) {
+        cols[fieldName] = index;
+      } //ignore first field, which just says "Foods"
     });
 
+    var lastMeal;
+    var meals = {
+      'breakfast': [],
+      'lunch': [],
+      'dinner': [],
+      'snacks': []
+    };
+
+    $('.first').each(function (index, element) {
```

Comment:
`$('first')` is not a very specific selector - it catches way more than just meal rows. Just search for the word "first" on this page: https://github.com/andrewzey/mfp/blob/master/test/mocks/diary-public.html#L154